src/repositories/consent.ts

```diff
@@ -58,6 +79,53 @@ export class FirebaseConsentRepository implements IConsentRepository {
         })
     })
   }
+
+  async updateConsentById(
+    id: string,
+    data: Record<string, any>
+  ): Promise<void> {
+    await firebase.firestore().collection('consents').doc(id).update(data)
+  }
+
+  async updateConsent(
+    conditions: Record<string, any>,
+    data: Record<string, any>
+  ): Promise<void> {
+    let firestoreQuery: FirebaseFirestore.Query = firebase
+      .firestore()
+      .collection('consents')
+
+    // Chain all of the given conditions to the query
+    for (const key in conditions) {
+      firestoreQuery = firestoreQuery.where(key, '==', conditions[key])
+    }
+
+    // Find and update all matching documents in Firebase that match the given conditions.
+    await firestoreQuery
```

Comment:
it isn't good to mix `await` with `then` style for handling asynchronous operations: there is a risk of misunderstanding of what is going on here. Is it really needed?

Reply:
doesn't seem like it
@stevenwjy - what do you think?

Reply:
oh yeah, sorry didn't notice about this... I think we can just remove the `await`? Does this occur in the code for the transfer repository as well? maybe can consider removing that ambiguity as well